ftdc/custom_format.go

```diff
@@ -318,19 +318,38 @@ func flatten(datum Datum, schema *schema) ([]float32, error) {
 	return ret, nil
 }
 
+// FlatDatum has the same information as a `datum`, but without the arbitrarily nested `Data`
+// map. Using dots to join keys as per the disk format. So where a `Data` map might be:
+//
+// { "webrtc": { "connections": 5, "bytesSent": 8096 } }
+//
+// A `FlatDatum` would be:
+//
+// [ Reading{"webrtc.connections", 5}, Reading{"webrtc.bytesSent", 8096} ]
+type FlatDatum struct {
+	Time     int64
+	Readings []Reading
+}
+
+// Reading is a "fully qualified" metric name paired with a value.
+type Reading struct {
+	MetricName string
+	Value      float32
+}
+
 // Parse reads the entire contents from `rawReader` and returns a list of `Datum`. If an error
 // occurs, the []Datum parsed up until the place of the error will be returned, in addition to a
 // non-nil error.
-func Parse(rawReader io.Reader) ([]Datum, error) {
+func Parse(rawReader io.Reader) ([]FlatDatum, error) {
```

Comment:
Turns out this is what I really wanted for parsing.

But this also broke test compiling (which prefer walking maps to make simple assertions).

I'm in the midst of fixing tests by adding a FlatDatum -> Datum converter

Reply:
> Turns out this is what I really wanted for parsing.

Sorry why is `FlatDatum` better than `Datum` for parsing?

Reply:
Sorry if this is pedantic: It's not better for parsing. It's better for rendering. Now I can just write a single loop:
```
for readings := range flatDatum.Readings {
  
}
```

instead of a nested:
```
for statsName, readings := range datum.Data {
  for _, reading := range readings {
    metricName := fmt.Sprintf("%s.%s", statsName, reading.MetricName)
  }
}
```

Put another way, with 1 graph per metric, the additional layer of structure doesn't help me with the output in any way. Maybe if we wanted to put a box around related graphs in a group. For example all of the "webrtc" metrics. Then the "string before the first dot" information provides utility.

Reply:
Gotcha; better for rendering tracks with my understanding. Thanks!